pkg/scheduler/queue/tenant_queues_test.go

```diff
@@ -32,55 +71,126 @@ func TestQueues(t *testing.T) {
 	assert.Nil(t, tenant)
 	assert.NoError(t, err)
 
-	// Add queues: [one]
-	qOne := getOrAdd(t, qb, "one", 0)
-	lastTenantIndex = confirmOrderForQuerier(t, qb, "querier-1", lastTenantIndex, qOne, qOne)
+	// Add tenant queues: tenant "one"
+	err = qb.tenantQuerierAssignments.createOrUpdateTenant("one", 0)
+	assert.NoError(t, err)
+
+	tenantOne := qb.tenantQuerierAssignments.tenantsByID["one"]
+
+	// Queue enough objects for tenant "one"
+	err = qb.enqueueObjectsForTests(tenantOne.tenantID, 10)
+	assert.NoError(t, err)
+	assert.NoError(t, isConsistent(qb))
+
+	//queuePathOne := QueuePath{"root", "one"}
+	expectedDequeueVals := []dequeueVal{
+		{buildExpectedObject(tenantOne.tenantID, 0), tenantOne},
+		{buildExpectedObject(tenantOne.tenantID, 1), tenantOne},
+	}
+	lastTenantIndex = assertExpectedValuesOnDequeue(t, qb, lastTenantIndex, "querier-1", expectedDequeueVals)
+
+	// Add tenant two
+	err = qb.tenantQuerierAssignments.createOrUpdateTenant("two", 0)
+	assert.NoError(t, err)
+
+	tenantTwo := qb.tenantQuerierAssignments.tenantsByID["two"]
+
+	err = qb.enqueueObjectsForTests(tenantTwo.tenantID, 10)
+	assert.NoError(t, err)
+	assert.NoError(t, isConsistent(qb))
+
+	expectedDequeueVals = []dequeueVal{
+		{buildExpectedObject(tenantTwo.tenantID, 0), tenantTwo},
+		{buildExpectedObject(tenantOne.tenantID, 2), tenantOne},
+		{buildExpectedObject(tenantTwo.tenantID, 1), tenantTwo},
+		{buildExpectedObject(tenantOne.tenantID, 3), tenantOne},
+	}
+
+	lastTenantIndex = assertExpectedValuesOnDequeue(t, qb, lastTenantIndex, "querier-1", expectedDequeueVals)
+
+	expectedDequeueVals = []dequeueVal{
+		{buildExpectedObject(tenantOne.tenantID, 4), tenantOne},
+		{buildExpectedObject(tenantTwo.tenantID, 2), tenantTwo},
+		{buildExpectedObject(tenantOne.tenantID, 5), tenantOne},
+	}
+	_ = assertExpectedValuesOnDequeue(t, qb, -1, "querier-2", expectedDequeueVals)
```

Comment:
lets put a little comment on why we are ignoring this lastTenantIndex return value, or just name the two vars like
`querier1LastTenantIndex` / `querier2LastTenantIndex`

Reply:
After getting through the rest of the test, I noticed that this is the last time we use `querier2` to dequeue.

I think we want to more explicitly demonstrate this idea that "different queriers will start at different points in the line, causing a nonlinear round-robin" as this is one of the aspects that is less immediately intuitive.

So we could either:
1. slightly downscope this test to only show the single-querier flow and just drop the other querier - move the demonstration of the line-jumping with lastTenantIndex to another test
2. do some more alternating between the two queriers in this test

I think option 2 is probably easier and keeps the test structure more intact